File: leetcode/226.翻转二叉树.cpp
/*
 * @Author: Opynicus
 * @Date: 2022-03-27 12:26:45
 * @LastEditTime: 2022-03-27 12:34:32
 * @LastEditors: Opynicus
 * @Description: leetcode binary tree problem
 * @FilePath: \undefinede:\JuniorWork\leetcode\226.翻转二叉树.cpp
 * 可以输入预定的版权声明、个性签名、空行等
 */
/*
 * @lc app=leetcode.cn id=226 lang=cpp
 *
 * [226] 翻转二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if (root == nullptr)
        {
            return NULL;
        }
        TreeNode* left = invertTree(root->left);
        TreeNode* right = invertTree(root->right);
        root->left = right;
        root->right = left;
        return root;
        
    }
};
// @lc code=end

File: leetcode/226.翻转二叉树.go
/*
 * @Author: Opynicus
 * @Date: 2022-03-28 17:15:15
 * @LastEditTime: 2022-03-28 17:18:11
 * @LastEditors: Opynicus
 * @Description: 
 * @FilePath: \leetcode\226.翻转二叉树.go
 * 可以输入预定的版权声明、个性签名、空行等
 */
/*
 * @lc app=leetcode.cn id=226 lang=golang
 *
 * [226] 翻转二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func invertTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil;
	}
	lchild := invertTree(root.Left)
	rchild := invertTree(root.Right)
	root.Left, root.Right = rchild, lchild
	return root
}
// @lc code=end

